ocs_ci/ocs/small_file_workload.py

```diff
@@ -1,141 +1,210 @@
+            samples (int): number of sample to run the test
+
+        """
+        self.crd_data["spec"]["workload"]["args"]["file_size"] = file_size
+        self.crd_data["spec"]["workload"]["args"]["files"] = files
+        self.crd_data["spec"]["workload"]["args"]["threads"] = threads
+        self.crd_data["spec"]["workload"]["args"]["samples"] = samples
+
+    def setup_vol_size(self, file_size, files, threads):
+        """
+        Calculating the size of the volume that need to be test, it should
+        be at least twice in the size then the size of the files, and at
+        least 100Gi.
+
+        Since the file_size is in Kb and the vol_size need to be in Gb, more
+        calculation is needed.
+
+        Args:
+            file_size (int): the file size in KB
+            files (int): number of file to use in the test
+            threads (int): number of threads to use in the test
+
+        """
+        vol_size = int(files * threads * file_size * 3)
+        vol_size = int(vol_size / constants.GB2KB)
+        if vol_size < 100:
+            vol_size = 100
+        self.crd_data["spec"]["workload"]["args"]["storagesize"] = f"{vol_size}Gi"
+
+    def setup_operations(self, ops):
+        """
+        Setting up the test operations
+
+        Args:
+            ops : can be list of operations or a string of one operation
+
+        """
+        if isinstance(ops, list):
+            self.crd_data["spec"]["workload"]["args"]["operation"] = ops
+        elif isinstance(ops, str):
+            self.crd_data["spec"]["workload"]["args"]["operation"] = [ops]
+
+    def run(self):
+        """
+        Run the benchmark and wait until it completed
+
+        """
+        # Create the benchmark object
+        self.sf_obj = OCS(**self.crd_data)
+        self.sf_obj.create()
+
+        # Wait for benchmark pods to get created - takes a while
+        for bench_pod in TimeoutSampler(
+            240,
+            10,
+            get_pod_name_by_pattern,
+            "smallfile-client",
+            benchmark_operator.BMO_NAME,
+        ):
+            try:
+                if bench_pod[0] is not None:
+                    small_file_client_pod = bench_pod[0]
+                    break
+            except IndexError:
+                log.info("Bench pod not ready yet")
+
+        bench_pod = OCP(kind="pod", namespace=benchmark_operator.BMO_NAME)
+        log.info("Waiting for SmallFile benchmark to Run")
+        assert bench_pod.wait_for_resource(
+            condition=constants.STATUS_RUNNING,
+            resource_name=small_file_client_pod,
+            sleep=30,
+            timeout=600,
+        )
+        log.info("The SmallFiles benchmark is running, wait for completion")
+        bench_pod.wait_for_resource(
+            condition=constants.STATUS_COMPLETED,
+            resource_name=small_file_client_pod,
+            timeout=3600,
+            sleep=60,
+        )
```

Comment:
Isn't it good to add assert here ? 
In-case of failure wait_for_resource function returns false, there are exceptions but don't think it will fail.

Reply:
the function wait_for_resource() is raising an exception in case of timeout reached and condition status not as expected